test/Dialect/FIRRTL/check-init-errors.mlir

```diff
@@ -0,0 +1,162 @@
+// RUN: circt-opt --pass-pipeline='builtin.module(firrtl.circuit(firrtl.module(firrtl-check-init)))' --split-input-file --verify-diagnostics %s
+
+// missing wire init
+firrtl.circuit "simplewire"   {
+  firrtl.module @simplewire() {
+  // expected-error @below {{Wire is not initialized.}}
+    %x = firrtl.wire  : !firrtl.uint<1>
+  }
+}
+
+// -----
+
+// missing output
+firrtl.circuit "simpleout"   {
+  // expected-error @below {{Port `a` is not initialized.}}
+  firrtl.module @simpleout(out %a: !firrtl.uint<1>) {
+  }
+}
+
+
+// -----
+
+// missing wire init
```

Comment:
Thank you for giving the circuits meaningful and unique (at least AFAICT) names! :heart: